snapshotstorageprovider/src/main/java/org/duracloud/snapshottask/snapshot/CreateSnapshotTaskRunner.java

```diff
@@ -230,26 +234,36 @@ protected String buildSnapshotProps(Map<String, String> props) {
      */
     protected String callBridge(RestHttpHelper restHelper,
                                 String snapshotURL,
-                                String snapshotBody) {
+                                String snapshotBody) throws Exception {
         log.info("Making SNAPSHOT call to URL {} with body {}",
                  snapshotURL, snapshotBody);
 
-        try {
             Map<String, String> headers = new HashMap<>();
             headers.put(HttpHeaders.CONTENT_TYPE, "application/json");
-            RestHttpHelper.HttpResponse response =
-                restHelper.put(snapshotURL, snapshotBody, headers);
+            RestHttpHelper.HttpResponse response = restHelper.put(snapshotURL, snapshotBody, headers);
             int statusCode = response.getStatusCode();
             if(statusCode != 200 && statusCode != 201) {
-                throw new RuntimeException("Unexpected response code: " +
-                                           statusCode);
+                String message = response.getResponseBody();
+
+                try {
+                    HashMap map = new JaxbJsonSerializer<HashMap>(HashMap.class).deserialize(message);
```

Comment:
It seems like this parsing of the bridge response message should be generic enough to live at a higher level, if not in another class, at least in its own method